### ISSUE TITLE
Problematic interactions with read logging in pgaudit

### ISSUE THREAD

**Issue by Farplaner** [timescale/timescaledb#1143]
**Relevant system information:**
 - OS: Red Hat Enterprise Linux 7.6
 - PostgreSQL version (output of `postgres --version`): 11.2
 - TimescaleDB version (output of `\dx` in `psql`): 1.2.2
 - Installation method: rpm installed from packagecloud.io

**Describe the bug**
I'm having issue migrating my data from another table to a newly created hypertable.  It is showing an error "cache lookup failed for type 4294967295" whenever I try to insert a greater number of rows (the threshold seems to be different for different data sets).

**To Reproduce**
1. Create test table:
```
db=# CREATE TABLE test (
db(#     id bigint NOT NULL,
db(#     text1 character varying(255),
db(#     text2 character varying(255),
db(#     text3 character varying(255),
db(#     time timestamp
db(# );
CREATE TABLE
db=# SELECT create_hypertable('test', 'time');
   create_hypertable
-----------------------
 (38,dbschema,test,t)
(1 row)
```
2. Insert 3 rows at once (works)
```
db=# INSERT INTO test
db-# values
db-# (1, 'test1', 'test2', 'test3', now() - interval '1' day),
db-# (2, 'test1', 'test2', 'test3', now() - interval '2' day),
db-# (3, 'test1', 'test2', 'test3', now() - interval '3' day);
INSERT 0 3
```
3. Insert 10 rows at once (fails)
```
db=# truncate table test;
TRUNCATE TABLE
db=# INSERT INTO test
db-# values
db-# (1, 'test1', 'test2', 'test3', now() - interval '1' day),
db-# (2, 'test1', 'test2', 'test3', now() - interval '2' day),
db-# (3, 'test1', 'test2', 'test3', now() - interval '3' day),
db-# (4, 'test1', 'test2', 'test3', now() - interval '4' day),
db-# (5, 'test1', 'test2', 'test3', now() - interval '5' day),
db-# (6, 'test1', 'test2', 'test3', now() - interval '6' day),
db-# (7, 'test1', 'test2', 'test3', now() - interval '7' day),
db-# (8, 'test1', 'test2', 'test3', now() - interval '8' day),
db-# (9, 'test1', 'test2', 'test3', now() - interval '9' day),
db-# (10, 'test1', 'test2', 'test3', now() - interval '10' day);
ERROR:  cache lookup failed for type 4294967295
CONTEXT:  SQL statement "SELECT *                       FROM _timescaledb_catalog.chunk c WHERE c.id = chunk_constraint_row.chunk_id"
PL/pgSQL function _timescaledb_internal.chunk_constraint_add_table_constraint(_timescaledb_catalog.chunk_constraint) line 9 at SQL statement
```

**Expected behavior**
Simply put, I expected the insert to work.

**Actual behavior**
It returns the error shown above

**Additional context**
This appears to happen when the dates are different.  I was able to insert a massive amount of data with the same dates with no problem:

```
db=# truncate table test;
TRUNCATE TABLE
db=# INSERT INTO test
db-# SELECT i, md5(random()::text), md5(random()::text), md5(random()::text), now()
db-# from generate_series(1, 100000) s(i);
INSERT 0 100000
```

**Comment by mfreed** [comment]
Hi @Farplaner Do you see similar problems without running TRUNCATE?

I'm having trouble reconstructing the same issue (on 1.2.2 instalelled through docker).
```
$ docker exec -it timescaledb psql -U postgres
psql (11.2)
Type "help" for help.

postgres=# CREATE TABLE test ( id bigint NOT NULL,
postgres(#  text1 character varying(255),
postgres(# text2 character varying(255),
postgres(# text3 character varying(255),
postgres(# time timestamp );
CREATE TABLE
postgres=# 
postgres=# INSERT INTO test
postgres-# values
postgres-# (1, 'test1', 'test2', 'test3', now() - interval '1' day),
postgres-# (2, 'test1', 'test2', 'test3', now() - interval '2' day),
postgres-# (3, 'test1', 'test2', 'test3', now() - interval '3' day);
INSERT 0 3
postgres=# 
postgres=# truncate table test;
TRUNCATE TABLE
postgres=# 
postgres=# INSERT INTO test
postgres-# values
postgres-# (1, 'test1', 'test2', 'test3', now() - interval '1' day),
postgres-# (2, 'test1', 'test2', 'test3', now() - interval '2' day),
postgres-# (3, 'test1', 'test2', 'test3', now() - interval '3' day),
postgres-# (4, 'test1', 'test2', 'test3', now() - interval '4' day),
postgres-# (5, 'test1', 'test2', 'test3', now() - interval '5' day),
postgres-# (6, 'test1', 'test2', 'test3', now() - interval '6' day),
postgres-# (7, 'test1', 'test2', 'test3', now() - interval '7' day),
postgres-# (8, 'test1', 'test2', 'test3', now() - interval '8' day),
postgres-# (9, 'test1', 'test2', 'test3', now() - interval '9' day),
postgres-# (10, 'test1', 'test2', 'test3', now() - interval '10' day);
INSERT 0 10
postgres=# 
postgres=# select * from test;
 id | text1 | text2 | text3 |            time            
----+-------+-------+-------+----------------------------
  1 | test1 | test2 | test3 | 2019-04-04 17:16:46.603687
  2 | test1 | test2 | test3 | 2019-04-03 17:16:46.603687
  3 | test1 | test2 | test3 | 2019-04-02 17:16:46.603687
  4 | test1 | test2 | test3 | 2019-04-01 17:16:46.603687
  5 | test1 | test2 | test3 | 2019-03-31 17:16:46.603687
  6 | test1 | test2 | test3 | 2019-03-30 17:16:46.603687
  7 | test1 | test2 | test3 | 2019-03-29 17:16:46.603687
  8 | test1 | test2 | test3 | 2019-03-28 17:16:46.603687
  9 | test1 | test2 | test3 | 2019-03-27 17:16:46.603687
 10 | test1 | test2 | test3 | 2019-03-26 17:16:46.603687
(10 rows)
```

**Comment by Farplaner** [comment]
Yes, it fails even without the truncate:

```
db=# drop table test;
DROP TABLE
db=#
db=# CREATE TABLE test (
db(#      id bigint NOT NULL,
db(#      text1 character varying(255),
db(#      text2 character varying(255),
db(#      text3 character varying(255),
db(#      time timestamp
db(# );
CREATE TABLE
db=#
db=# SELECT create_hypertable('test', 'time');
   create_hypertable
-----------------------
 (39,tadschema,test,t)
(1 row)

db=#
db=# INSERT INTO test
db-# values
db-# (1, 'test1', 'test2', 'test3', now() - interval '1' day),
db-# (2, 'test1', 'test2', 'test3', now() - interval '2' day),
db-# (3, 'test1', 'test2', 'test3', now() - interval '3' day),
db-# (4, 'test1', 'test2', 'test3', now() - interval '4' day),
db-# (5, 'test1', 'test2', 'test3', now() - interval '5' day),
db-# (6, 'test1', 'test2', 'test3', now() - interval '6' day),
db-# (7, 'test1', 'test2', 'test3', now() - interval '7' day),
db-# (8, 'test1', 'test2', 'test3', now() - interval '8' day),
db-# (9, 'test1', 'test2', 'test3', now() - interval '9' day),
db-# (10, 'test1', 'test2', 'test3', now() - interval '10' day);
ERROR:  cache lookup failed for type 4294967295
CONTEXT:  SQL statement "SELECT *                       FROM _timescaledb_catalog.chunk c WHERE c.id = chunk_constraint_row.chunk_id"
PL/pgSQL function _timescaledb_internal.chunk_constraint_add_table_constraint(_timescaledb_catalog.chunk_constraint) line 9 at SQL statement
```

I'm at a loss at the moment.  If there are other details I can look for I'll gladly find out.

**Comment by mfreed** [comment]
Is this a brand new database?  4294967295 looks like int max... 🤔

**Comment by Farplaner** [comment]
It was not a brand new database, but just for testing I went ahead and created a new database in the same box and tried it (the following is after connecting to the new database (mydb):

```
mydb=# create schema myschema;
CREATE SCHEMA
mydb=# SET search_path TO myschema;
SET
mydb=# SET statement_timeout = '1h';
SET
mydb=# SHOW statement_timeout;
 statement_timeout
-------------------
 1h
(1 row)

mydb=#
mydb=# CREATE EXTENSION timescaledb CASCADE;
CREATE EXTENSION
mydb=# CREATE TABLE test (
mydb(#      id bigint NOT NULL,
mydb(#      text1 character varying(255),
mydb(#      text2 character varying(255),
mydb(#      text3 character varying(255),
mydb(#      time timestamp
mydb(# );
CREATE TABLE
mydb=#
mydb=# SELECT create_hypertable('test', 'time');
  create_hypertable
---------------------
 (1,myschema,test,t)
(1 row)

mydb=# INSERT INTO test
mydb-# values
mydb-# (1, 'test1', 'test2', 'test3', now() - interval '1' day),
mydb-# (2, 'test1', 'test2', 'test3', now() - interval '2' day),
mydb-# (3, 'test1', 'test2', 'test3', now() - interval '3' day),
mydb-# (4, 'test1', 'test2', 'test3', now() - interval '4' day),
mydb-# (5, 'test1', 'test2', 'test3', now() - interval '5' day),
mydb-# (6, 'test1', 'test2', 'test3', now() - interval '6' day),
mydb-# (7, 'test1', 'test2', 'test3', now() - interval '7' day),
mydb-# (8, 'test1', 'test2', 'test3', now() - interval '8' day),
mydb-# (9, 'test1', 'test2', 'test3', now() - interval '9' day),
mydb-# (10, 'test1', 'test2', 'test3', now() - interval '10' day);
ERROR:  cache lookup failed for type 4294967295
CONTEXT:  SQL statement "SELECT *                       FROM _timescaledb_catalog.chunk c WHERE c.id = chunk_constraint_row.chunk_id"
PL/pgSQL function _timescaledb_internal.chunk_constraint_add_table_constraint(_timescaledb_catalog.chunk_constraint) line 9 at SQL statement
```

**Comment by mfreed** [comment]
Sigh.

```
postgres=# create schema myschema;
CREATE SCHEMA
postgres=# SET search_path TO myschema;
SET
postgres=# SET statement_timeout = '1h';
SET
postgres=# SHOW statement_timeout;
 statement_timeout 
-------------------
 1h
(1 row)

postgres=# 
postgres=# 
postgres=# CREATE EXTENSION timescaledb CASCADE;
WARNING:   WELCOME TO
...
CREATE EXTENSION
postgres=# CREATE TABLE test ( id bigint NOT NULL,
postgres(#  text1 character varying(255),
postgres(# text2 character varying(255),
postgres(# text3 character varying(255),
postgres(# time timestamp );
CREATE TABLE

postgres=# SELECT create_hypertable('test', 'time');
NOTICE:  adding not-null constraint to column "time"
DETAIL:  Time dimensions cannot have NULL values
  create_hypertable  
---------------------
 (1,myschema,test,t)
(1 row)

postgres=# INSERT INTO test
postgres-# values
postgres-# (1, 'test1', 'test2', 'test3', now() - interval '1' day),
postgres-# (2, 'test1', 'test2', 'test3', now() - interval '2' day),
postgres-# (3, 'test1', 'test2', 'test3', now() - interval '3' day),
postgres-# (4, 'test1', 'test2', 'test3', now() - interval '4' day),
postgres-# (5, 'test1', 'test2', 'test3', now() - interval '5' day),
postgres-# (6, 'test1', 'test2', 'test3', now() - interval '6' day),
postgres-# (7, 'test1', 'test2', 'test3', now() - interval '7' day),
postgres-# (8, 'test1', 'test2', 'test3', now() - interval '8' day),
postgres-# (9, 'test1', 'test2', 'test3', now() - interval '9' day),
postgres-# (10, 'test1', 'test2', 'test3', now() - interval '10' day);
INSERT 0 10
postgres=# 
```

Anyway, will see about digging more into code or trying to replicate your install.  Thanks.

**Comment by Farplaner** [comment]
I completely reset the postgres installation with `/usr/pgsql-11/bin/postgresql-11-setup initdb` after deleting the data directory, and then did the same test above.  It worked.

So looks like it's something in my postgres installation that is causing it to fail.  I'll talk to my admin to see what is the difference from the base postgres install.  Hopefully I can provide more details so we can get to the bottom of this.

**Comment by Farplaner** [comment]
Looks like the conflict is with pgaudit.  I went back to the previous snapshot where I have this issue and unloaded the pgaudit extension, and everything works again.  We're using version pgaudit 1.3.0 on our servers.

**Comment by Farplaner** [comment]
Looks like I have it nailed down to this:
```
pgaudit.log='ddl,write,role'
#pgaudit.log='ddl,write,role,read'
```
Somehow removing the read logging was the trick.

**Comment by mfreed** [comment]
Thanks for the report, @Farplaner !

**Comment by svenklemm** [comment]
Closing this since it seems it was resolved. Feel free to reopen if the issue is still relevant.
